### PR TITLE
protonvpn-gui: 4.3.0 -> 4.4.4

### PR DESCRIPTION
**Pull request by SebTM**
## Description of changes

Update package, add two new libraries, add nix-update-script for all (will work after https://github.com/Mic92/nix-update/pull/246#issuecomment-2289217233 is fixed), nixfmt-rfc-style

This release includes experimental wireguard support for the new desktop-client.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SebTM** [comment]
Result of `nixpkgs-review pr 334674` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>54 packages built:</summary>
  <ul>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>python311Packages.proton-core</li>
    <li>python311Packages.proton-core.dist</li>
    <li>python311Packages.proton-keyring-linux</li>
    <li>python311Packages.proton-keyring-linux-secretservice</li>
    <li>python311Packages.proton-keyring-linux-secretservice.dist</li>
    <li>python311Packages.proton-keyring-linux.dist</li>
    <li>python311Packages.proton-vpn-api-core</li>
    <li>python311Packages.proton-vpn-api-core.dist</li>
    <li>python311Packages.proton-vpn-connection</li>
    <li>python311Packages.proton-vpn-connection.dist</li>
    <li>python311Packages.proton-vpn-killswitch</li>
    <li>python311Packages.proton-vpn-killswitch-network-manager</li>
    <li>python311Packages.proton-vpn-killswitch-network-manager-wireguard</li>
    <li>python311Packages.proton-vpn-killswitch-network-manager-wireguard.dist</li>
    <li>python311Packages.proton-vpn-killswitch-network-manager.dist</li>
    <li>python311Packages.proton-vpn-killswitch.dist</li>
    <li>python311Packages.proton-vpn-logger</li>
    <li>python311Packages.proton-vpn-logger.dist</li>
    <li>python311Packages.proton-vpn-network-manager</li>
    <li>python311Packages.proton-vpn-network-manager-openvpn</li>
    <li>python311Packages.proton-vpn-network-manager-openvpn.dist</li>
    <li>python311Packages.proton-vpn-network-manager-wireguard</li>
    <li>python311Packages.proton-vpn-network-manager-wireguard.dist</li>
    <li>python311Packages.proton-vpn-network-manager.dist</li>
    <li>python311Packages.proton-vpn-session</li>
    <li>python311Packages.proton-vpn-session.dist</li>
    <li>python312Packages.proton-core</li>
    <li>python312Packages.proton-core.dist</li>
    <li>python312Packages.proton-keyring-linux</li>
    <li>python312Packages.proton-keyring-linux-secretservice</li>
    <li>python312Packages.proton-keyring-linux-secretservice.dist</li>
    <li>python312Packages.proton-keyring-linux.dist</li>
    <li>python312Packages.proton-vpn-api-core</li>
    <li>python312Packages.proton-vpn-api-core.dist</li>
    <li>python312Packages.proton-vpn-connection</li>
    <li>python312Packages.proton-vpn-connection.dist</li>
    <li>python312Packages.proton-vpn-killswitch</li>
    <li>python312Packages.proton-vpn-killswitch-network-manager</li>
    <li>python312Packages.proton-vpn-killswitch-network-manager-wireguard</li>
    <li>python312Packages.proton-vpn-killswitch-network-manager-wireguard.dist</li>
    <li>python312Packages.proton-vpn-killswitch-network-manager.dist</li>
    <li>python312Packages.proton-vpn-killswitch.dist</li>
    <li>python312Packages.proton-vpn-logger</li>
    <li>python312Packages.proton-vpn-logger.dist</li>
    <li>python312Packages.proton-vpn-network-manager</li>
    <li>python312Packages.proton-vpn-network-manager-openvpn</li>
    <li>python312Packages.proton-vpn-network-manager-openvpn.dist</li>
    <li>python312Packages.proton-vpn-network-manager-wireguard</li>
    <li>python312Packages.proton-vpn-network-manager-wireguard.dist</li>
    <li>python312Packages.proton-vpn-network-manager.dist</li>
    <li>python312Packages.proton-vpn-session</li>
    <li>python312Packages.proton-vpn-session.dist</li>
  </ul>
</details>

**Review by pbsds — COMMENTED** [review]
Please separate this into separate bump, adoption and init commits. I leave it up to you whether to separate out the reformats as well or not.

**Comment by nyabinary** [comment]
> Please separate this into separate bump, adoption and init commits. I leave it up to you whether to separate out the reformats as well or not.

Also, shouldn't all the new packages be in by-name?

**Comment by SebTM** [comment]
> Also, shouldn't all the new packages be in by-name?

Only the new one or should I move the existing one as well?

**Comment by nyabinary** [comment]
> > Also, shouldn't all the new packages be in by-name?
> 
> Only the new one or should I move the existing one as well?

Isn't required to move existing ones but might as well imo.

**Comment by pbsds** [comment]
There is supposed to be a by-name treewide migration soon^tm but it has been postponed due to all the other stuff going on. If you want to be able to invoke the nixpkgs merge bot on r-ryantm bump PRs, then you can migrate your packages to by-name, otherwise there is no need.

**Comment by Steinhagen** [comment]
Is there any update for this? Gave this PR a try and the update works as expected.

**Comment by natsukium** [comment]
result of `nixpkgs-lint --include-unfinished-lints pkgs/development/python-modules/{proton-vpn-killswitch-network-manager-wireguard,proton-vpn-network-manager-wireguard}` [1](https://github.com/natsukium/nixpkgs-lint/tree/620731fc7ec66b0c67e6b8ea039545f71ea78409)
```
Advice: python package in nativeBuildInputs
    ╭─[pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix:33:5]
    │
 30 │ ╭─▶   nativeBuildInputs = [
    ┆ ┆   
 33 │ │       setuptools
    │ │       ─────┬────  
    │ │            ╰────── move this from nativeBuildInputs to build-system
 34 │ ├─▶   ];
    │ │          
    │ ╰────────── part of this list
────╯
Advice: python package in nativeBuildInputs
    ╭─[pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix:28:5]
    │
 25 │ ╭─▶   nativeBuildInputs = [
    ┆ ┆   
 28 │ │       setuptools
    │ │       ─────┬────  
    │ │            ╰────── move this from nativeBuildInputs to build-system
 29 │ ├─▶   ];
    │ │          
    │ ╰────────── part of this list
────╯
```

File: pkgs/applications/networking/protonvpn-gui/default.nix
Comment (natsukium) [inline pkgs/applications/networking/protonvpn-gui/default.nix:98]:
```suggestion
```
We don't need to set such a simple update-script because the python builder has its own update-script.
Please apply to all other changes.
https://github.com/NixOS/nixpkgs/blob/00f536f650269d876d5a7c8dcc8bb567ee4df233/pkgs/development/interpreters/python/mk-python-derivation.nix#L218-L224

File: pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix:36]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix:48]: How about using `pytest-cov-stub`?

File: pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix:31]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix:41]: Please enable tests with `pytestCheckHook` and `pythonImportsCheck`.

**Review by natsukium — DISMISSED** [review]

File: pkgs/applications/networking/protonvpn-gui/default.nix
Comment (SebTM) [reply pkgs/applications/networking/protonvpn-gui/default.nix:98]: But why aren't there any PRs for the existing updates triggered?

**Comment by nyabinary** [comment]
bump cc: @SebTM :p

File: pkgs/applications/networking/protonvpn-gui/default.nix
Comment (SebTM) [reply pkgs/applications/networking/protonvpn-gui/default.nix:98]: This appears to be only working for python-libs that are published via some kind of dependency-hub not only via github - can you confirm?

For me I need this that the `maintainers/scripts/update.nix` recognizes new releases on GitHub for the package e.g. `nix-shell maintainers/scripts/update.nix --argstr package python3Packages.proton-vpn-api-core`

Comment (natsukium) [reply pkgs/applications/networking/protonvpn-gui/default.nix:98]: Sorry, I noticed that python-update-libraries checks github releases but not tags.
That's why it did not update this package.

https://github.com/NixOS/nixpkgs/blob/3c799ef618f8a0ea383898b7e8877fcec4783988/pkgs/development/interpreters/python/update-python-libraries/update-python-libraries.py#L265-L267

I will fix this, but it's not a blocker.

Comment (SebTM) [reply pkgs/applications/networking/protonvpn-gui/default.nix:98]: Thanks for letting me know and taking care of this - will remove the update-scripts then :)

File: pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix
Comment (SebTM) [reply pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix:32]: Should this be done to all modules then? Can we do this in a one-commit follow-up please it was already a bunch of work to split this :/

**Comment by SebTM** [comment]
@natsukium `Advice: python package in nativeBuildInputs` where should they go? I just inherited this part

**Comment by SebTM** [comment]
@nyabinary Always feel free to lend a hand and take over if you need/depend on something I might have started 🙏🏻

File: pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix
Comment (SebTM) [reply pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix:36]: See below - can do it for all of protonvpn-gui's packages in this MR as one additional commit or as follow-up PR

**Comment by Steinhagen** [comment]
Just gave this a try and from a functionality perspective, everything works as expected.

**Comment by Steinhagen** [comment]
Is there anything left that is blocking this PR? I retried this today and everything worked as expected.
Except for those 2 open questions that can be addressed later, I think this PR is good enough for a package update.

**Comment by SebTM** [comment]
> Is there anything left that is blocking this PR?
Merge is blocked by @natsukium change requested

I would like to have two approvals to merge it myself.

File: pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix
Comment (pbsds) [reply pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix:32]: in general yes. If a review comment points out a systemic issue it is best to fix all occourances. Most reviewers don't have the bandwith to go thrugh the entire diff and mark each instance.

File: pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix
Comment (pbsds) [inline pkgs/development/python-modules/proton-vpn-killswitch-network-manager-wireguard/default.nix:34]:
```suggestion
  nativeBuildInputs = [
    # Solves ImportError: cannot import name NM, introspection typelib not found
    gobject-introspection
  ];
  
  build-system [
    setuptools
  ];
```

File: pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix
Comment (pbsds) [inline pkgs/development/python-modules/proton-vpn-network-manager-wireguard/default.nix:29]: same here

**Review by pbsds — COMMENTED** [review]
> where should they go? I just inherited this parts

see suggestions

> I would like to have two approvals to merge it myself.

I love this level caution

**Comment by SebTM** [comment]
I've resolved the review comments - but I can't get it building like this somehow the tests on "protonvpn-gui" will be executed despite `doCheck` is false or do I miss something?

Maybe someone can help out here :)

Is there a way to keep these as separate commit as splitting it up to the 15 commits with new files is quite a mess and does not even work well with `git stash`?

File: pkgs/development/python-modules/proton-core/default.nix
Comment (pbsds) [inline pkgs/development/python-modules/proton-core/default.nix:40]: This is why pytestCheckPhase is being run:

```suggestion
  nativeCheckInputs = [
```

File: pkgs/development/python-modules/proton-core/default.nix
Comment (pbsds) [inline pkgs/development/python-modules/proton-core/default.nix:30]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/proton-core/default.nix
Comment (SebTM) [reply pkgs/development/python-modules/proton-core/default.nix]: 🤦🏻‍♂️ Thanks 👍🏻

**Review by Steinhagen — APPROVED** [review]
Everything looks and works as expected! Nice job!

**Review by pbsds — APPROVED** [review]

**Comment by pbsds** [comment]
the ofborg darwin queue was resently wiped, lets wait for ci

**Pull request merged**
